mediapipe/tasks/ios/test/vision/object_detector/MPPObjectDetectorTests.m

```diff
@@ -0,0 +1,703 @@
+  options.categoryDenylist = @[ @"cat" ];
+
+  MPPObjectDetector *objectDetector = [self objectDetectorWithOptionsSucceeds:options];
+
+  NSArray<MPPDetection *> *detections = @[
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1
+                                   score:0.476562f
+                            categoryName:@"teddy bear"
+                             displayName:nil],
+    ]
+                                 boundingBox:CGRectMake(780, 407, 314, 190)
+                                   keypoints:nil],
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1
+                                   score:0.390625f
+                            categoryName:@"teddy bear"
+                             displayName:nil],
+    ]
+                                 boundingBox:CGRectMake(90, 225, 568, 366)
+                                   keypoints:nil],
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1
+                                   score:0.367188f
+                            categoryName:@"teddy bear"
+                             displayName:nil],
+    ]
+                                 boundingBox:CGRectMake(888, 434, 187, 167)
+                                   keypoints:nil],
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1 score:0.332031f categoryName:@"bed" displayName:nil],
+    ]
+                                 boundingBox:CGRectMake(79, 364, 1097, 224)
+                                   keypoints:nil],
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1
+                                   score:0.289062f
+                            categoryName:@"teddy bear"
+                             displayName:nil],
+    ]
+                                 boundingBox:CGRectMake(605, 398, 445, 199)
+                                   keypoints:nil],
+  ];
+
+  MPPObjectDetectionResult *expectedDetectionResult =
+      [[MPPObjectDetectionResult alloc] initWithDetections:detections timestampMs:0];
+
+  [self assertResultsOfDetectInImageWithFileInfo:kCatsAndDogsImage
+                             usingObjectDetector:objectDetector
+                                      maxResults:-1
+                     equalsObjectDetectionResult:expectedDetectionResult];
+}
+
+- (void)testDetectWithOrientationSucceeds {
+  MPPObjectDetectorOptions *options = [self objectDetectorOptionsWithModelName:kModelName];
+  options.maxResults = 1;
+
+  MPPObjectDetector *objectDetector = [self objectDetectorWithOptionsSucceeds:options];
+
+  NSArray<MPPDetection *> *detections = @[
+    [[MPPDetection alloc] initWithCategories:@[
+      [[MPPCategory alloc] initWithIndex:-1
+                                   score:0.750000f
+                            categoryName:@"teddy bear"
```

Comment:
There is a bug in how we handle image rotation in all of our tasks. We are going through the APIs and fixing this. Can you see if you can port https://github.com/google/mediapipe/commit/e3185e3df05dc0ef347118c18e08b8a9ff5263d4 and verify that we are getting a "cat" here?